zebra-chain/src/error.rs

```diff
@@ -0,0 +1,38 @@
+//! Errors that can occur inside any `zebra-chain` submodule.
+use thiserror::Error;
+
+/// Number of times a `diversify_hash` will try to obtain a diversified base point.
+pub const DIVERSIFY_HASH_TRIES: u8 = 5;
+
+/// Errors related to random bytes generation.
+#[derive(Error, Copy, Clone, Debug, PartialEq, Eq)]
+pub enum RandError {
+    /// Error of the `try_fill_bytes` function.
+    #[error("failed to generate a secure stream of random bytes")]
+    FillBytes,
+    /// Error of the `diversify_hash` function.
+    #[error(
+        "failed to generate a diversified base point after {} tries",
+        DIVERSIFY_HASH_TRIES
+    )]
+    DiversifyHash,
+}
+
+/// Errors related to `AffinePoint`.
+#[derive(Error, Copy, Clone, Debug, PartialEq, Eq)]
+pub enum AffineError {
```

Comment:
###### Optional

I'd go with `AffinePointError` here. 


```suggestion
pub enum AffinePointError {
```

Reply:
Resolved in [421c9b8](https://github.com/ZcashFoundation/zebra/pull/6385/commits/421c9b86d6cacd1de6c4a73332ab29a175998d07).